recipes/physx/4.1.1/conanfile.py

```diff
@@ -0,0 +1,264 @@
+import os
+import time
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+class PhysXConan(ConanFile):
+    name = "physx"
+    description = "The NVIDIA PhysX SDK is a scalable multi-platform " \
+                  "physics solution supporting a wide range of devices, " \
+                  "from smartphones to high-end multicore CPUs and GPUs."
+    license = "BSD-3-Clause"
+    topics = ("conan", "PhysX", "physics")
+    homepage = "https://github.com/NVIDIAGameWorks/PhysX"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
+    settings = "os", "compiler", "arch", "build_type"
+    short_paths = True
+    options = {
+        "shared": [True, False],
+        "release_build_type": ["profile", "release"],
+        "enable_simd": [True, False],
+        "enable_float_point_precise_math": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": True,
+        "release_build_type": "release",
+        "enable_simd": True,
+        "enable_float_point_precise_math": False,
+        "fPIC": True,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        del self.options.fPIC # fpic is managed by physx build process (in physx/source/compiler/cmake/CMakeLists.txt)
```

Comment:
Right now, the `fPIC` option is useless.
You can fix this by a small patch that removes the line
https://github.com/NVIDIAGameWorks/PhysX/blob/ae80dede0546d652040ae6260a810e53e20a06fa/physx/source/compiler/cmake/CMakeLists.txt#L122
Then you can remove `fPIC` on Windows in `config_options` and when `shared` in `configure`.

Reply:
In this CMakeLists, this line is also useless, because fPIC is again forced for each target. Therefore the patch should be more complex. Moreover, when build as shared library, it has to be noticed that several libs are still static and linked against shared libs.

Reply:
Using static libraries as intermediate libraries is not necessarily bad as they can also contain fPIC-enabled objects. But indeed, making all the libraries in physx comply with conan's options may require some more complex patching.

Reply:
@madebr, do you want to propose a patch or may I resolve this conversation?

Reply:
See https://github.com/SpaceIm/conan-center-index/pull/4

Reply:
I'm wondering:
If conan builds PhysX as a shared lib: with this modification, I understrand that PhysXPvdSDK will be built as a static lib without fPIC and linked into shared libs PhysXExtensions, PhysX and PhysXVehicle...

Reply:
Good point, I just tried and the build fails. I now get your issue about static library dependencies not being fPIC.
I pushed a new patch that should fix this.

Reply:
Ok I see, thanks.
If -Werror is removed, we can also remove patches 0004-Gcc-Wrestrict-Wclass-memaccess-workaround.patch and 0005-Macos-Wno-atomic-implicit-seq-cst.patch

Reply:
merged + patches related to compilers warnings removed, thanks ;)

Reply:
Actually, there is no need to introduce PHYSX_CONAN_FPIC.
It is sufficient to just replace lines referencing CMAKE_POSITION_INDEPENDENT_CODE (in PhysX CMake files) by empty strings.
Therefore, we can just rely on conan to properly set CONAN_CMAKE_POSITION_INDEPENDENT_CODE depending on options (as I understand source code of conan, self.options.fPIC must not be removed in configure if shared).

Reply:
Indeed, that will work as well.